modules/indexing/src/test/java/org/apache/ignite/cache/query/IndexQueryCacheKeyValueFieldsTest.java

```diff
@@ -0,0 +1,120 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.cache.query;
+
+import java.util.LinkedHashMap;
+import java.util.Random;
+import javax.cache.Cache;
+
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteDataStreamer;
+import org.apache.ignite.cache.QueryEntity;
+import org.apache.ignite.cache.QueryIndex;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.util.typedef.F;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+import static org.apache.ignite.cache.query.IndexQueryCriteriaBuilder.gt;
+
+/** The test checks that IndexQuery correctly works with indexes contain single field: cache key or value. */
+public class IndexQueryCacheKeyValueFieldsTest extends GridCommonAbstractTest {
+    /** */
+    private static final String CACHE = "TEST_CACHE";
+
+    /** */
+    private static final int CNT = 10_000;
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        Ignite crd = startGrids(2);
+
+        try (IgniteDataStreamer<String, Integer> streamer = crd.dataStreamer(CACHE)) {
+            for (int i = 0; i < CNT; i++)
+                streamer.addData(key(i), i);
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String instanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(instanceName);
+
+        // Cache with signle value field.
+        LinkedHashMap<String, String> fields = new LinkedHashMap<>();
+        fields.put("f1", "java.lang.String");
+        fields.put("f2", "java.lang.Integer");
+
+        QueryEntity qryEntity = new QueryEntity()
+            .setKeyFieldName("f1")
+            .setValueFieldName("f2")
+            .setTableName("TEST")
+            .setKeyType(String.class.getName())
+            .setValueType(Integer.class.getName())
+            .setFields(fields)
+            .setIndexes(F.asList(
+                new QueryIndex("f1"),
+                new QueryIndex("f2"),
+                new QueryIndex("_VAL")));
+
+        return cfg.setCacheConfiguration(
+            new CacheConfiguration<String, Integer>("TEST_CACHE")
+                .setQueryEntities(F.asList(qryEntity)));
+    }
+
+    /** */
+    @Test
+    public void testValueAliasField() {
+        check("f2", false);
+    }
+
+    /** */
+    @Test
+    public void testValueField() {
+        check("_VAL", false);
+    }
+
+    /** */
+    @Test
+    public void testKeyAliasField() {
+        check("f1", true);
+    }
+
+    /** */
+    @Test
+    public void testKeyField() {
```

Comment:
Let's also add tests with upper/lower case letters

Reply:
Added tests for cases and escaping.